Rocket.ChatTests/Controllers/ChatDataControllerSpec.swift

```diff
@@ -34,6 +34,47 @@ class ChatDataControllerSpec: XCTestCase {
         XCTAssertEqual(indexPaths.count, 2, "indexPaths will have two results")
     }
 
+    func testIndexPathOf() {
+        let controller = ChatDataController()
+
+        var obj2 = ChatData(type: .message, timestamp: Date())
+        obj2.timestamp = Date().addingTimeInterval(1.0)
+
+        var obj1 = ChatData(type: .message, timestamp: Date())
+        obj1.timestamp = Date()
+
+        var obj3 = ChatData(type: .message, timestamp: Date())
+        obj3.timestamp = Date().addingTimeInterval(2.0)
+
+        controller.insert([obj2, obj1, obj3])
+
+        XCTAssertEqual(controller.indexPathOf(obj1.identifier)?.row, 1, "obj0 found in correct row")
```

Comment:
Why is the first one row `1`? Imagining that's because of the loader or the date division... You can use the `ChatDataController.loadedAllMessages` property to avoid that.